pyrokinetics/equilibrium.py

```diff
@@ -159,6 +162,210 @@ def read_geqdsk(self, psi_n_lcfs=1.0):
+                np.cos(i * theta) * data[f"YMC{i:02d}"][time_index, :] * 1e-2
+            )
+
+            # TRANSP doesn't stored 0th sin moment = 0.0 by defn
+            if i == 0:
+                R_mom_sin[i, :, :] = 0.0
+                Z_mom_sin[i, :, :] = 0.0
+            else:
+                R_mom_sin[i, :, :] = (
+                    np.sin(i * theta) * data[f"RMS{i:02d}"][time_index, :] * 1e-2
+                )
+                Z_mom_sin[i, :, :] = (
+                    np.sin(i * theta) * data[f"YMS{i:02d}"][time_index, :] * 1e-2
+                )
+
+        Rsur = np.sum(R_mom_cos, axis=0) + np.sum(R_mom_sin, axis=0)
+        Zsur = np.sum(Z_mom_cos, axis=0) + np.sum(Z_mom_sin, axis=0)
+
+        psi_axis = data["PSI0_TR"][time_index]
+        psi_bdry = data["PLFLXA"][time_index]
+
+        current = data["PCUR"][time_index]
+
+        # Load in 1D profiles
+        q = data["Q"][time_index, :]
+        press = data["PMHD_IN"][time_index, :]
+
+        # F is on a different grid and need to remove duplicated HFS points
+        psi_rmajm = data["PLFMP"][time_index, :]
+        rmajm_ax = np.argmin(psi_rmajm)
+        psi_n_rmajm = psi_rmajm[rmajm_ax:] / psi_rmajm[-1]
+
+        # f = (Bt / |B|) * |B| *  R
+        f = (
+            data["FBTX"][time_index, rmajm_ax:]
+            * data["BTX"][time_index, rmajm_ax:]
+            * data["RMAJM"][time_index, rmajm_ax:]
+            * 1e-2
+        )
+
+        psi = data["PLFLX"][time_index, :]
+        psi_n = psi / psi[-1]
+
+        rbdry = Rsur[:, -1]
+        zbdry = Zsur[:, -1]
+
+        # Default to netCDF values if None
+        if Rmin is None:
+            Rmin = min(rbdry)
+
+        if Rmax is None:
+            Rmax = max(rbdry)
+
+        if Zmin is None:
+            Zmin = min(zbdry)
+
+        if Zmax is None:
+            Zmax = max(zbdry)
+
+        if nr is None:
+            nr = nradial
+
+        if nz is None:
+            nz = nradial
+
+        Rgrid = np.linspace(Rmin, Rmax, nr)
+        Zgrid = np.linspace(Zmin, Zmax, nz)
+
+        # Set up 1D profiles
+
+        # Using interpolated splines
+        q_interp = InterpolatedUnivariateSpline(psi_n, q)
+        press_interp = InterpolatedUnivariateSpline(psi_n, press)
+        f_interp = InterpolatedUnivariateSpline(psi_n_rmajm, f)
+        f2_interp = InterpolatedUnivariateSpline(psi_n_rmajm, f**2)
+        ffprime_interp = f2_interp.derivative()
+
+        # Set up 2D profiles
+        # Re-map from R(theta, psi), Z (theta, psi) to psi(R, Z)
+        Rmesh, Zmesh = np.meshgrid(Rgrid, Zgrid)
+        Rmesh_flat = np.ravel(Rmesh)
+        Zmesh_flat = np.ravel(Zmesh)
+
+        Rflat = np.ravel(Rsur.T)
+        Zflat = np.ravel(Zsur.T)
+        psiflat = np.repeat(psi, ntheta)
+
+        RZflat = np.stack([Rflat, Zflat], -1)
+        RZmesh_flat = np.stack([Rmesh_flat, Zmesh_flat], -1)
+
+        # Interpolate using flat data
+        psiRZ_interp = RBFInterpolator(RZflat, psiflat, kernel="cubic")
```

Comment:
```suggestion
        psiRZ_interp = RBFInterpolator(RZflat, psiflat, kernel="cubic", neighbors=5)
```

This seems to fix the intermittent test failures on non-square number of OpenMP threads, with the added bonus of being _significantly_ faster: 40s vs 4s.

By default, the `RBFInterpolator` uses all of the points to construct the radial basis functions. From the [docs](https://docs.scipy.org/doc/scipy//reference/generated/scipy.interpolate.RBFInterpolator.html#scipy.interpolate.RBFInterpolator):

> The memory required to solve for the RBF interpolation coefficients increases quadratically with the number of data points, which can become impractical when interpolating more than about a thousand data points. To overcome memory limitations for large interpolation problems, the neighbors argument can be specified to compute an RBF interpolant for each evaluation point using only the nearest data points.

Reply:
Very nice spot there!